presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/TopologyAwareNodeSelector.java

```diff
@@ -123,9 +131,10 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
         Set<NetworkLocation> filledLocations = new HashSet<>();
         Set<InternalNode> blockedExactNodes = new HashSet<>();
         boolean splitWaitingForAnyNode = false;
+        List<HostAddress> sortedCandidates = NodeSelector.sortedNodes(nodeMap);
         for (Split split : splits) {
-            if (!split.isRemotelyAccessible()) {
-                List<InternalNode> candidateNodes = selectExactNodes(nodeMap, split.getAddresses(), includeCoordinator);
+            if (split.getNodeSelectionStrategy() != NO_PREFERENCE) {
```

Comment:
This is wrong, we should say `split.getNodeSelectionStrategy() == HARD_AFFINITY`